crates/driver/src/settlement_proposal.rs

```diff
@@ -243,6 +245,13 @@ impl SettlementProposal {
             gas_used += GAS_PER_ORDER.into();
         }
 
+        for (token, balance_before) in contract_buffer {
+            if !matches!(balances.get(token), Some(balance_after) if balance_after >= balance_before)
+            {
+                anyhow::bail!("solution would drain settlement contract buffers illegally");
+            }
```

Comment:
This can be abridged to:
```suggestion
            ensure!(
                matches!(balances.get(token), Some(balance_after) if balance_after >= balance_before),
                "solution would drain settlement contract buffers illegally",
            );
```